test/simplify_algebra_test.cpp

```diff
@@ -4133,6 +4133,167 @@ TEST_CASE(dot_slice_a)
+        auto three = m1.add_literal(
+            migraphx::generate_literal({migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+
+        auto add1 = m1.add_instruction(migraphx::make_op("add"), slice_a, one);
+        auto add2 = m1.add_instruction(migraphx::make_op("add"), slice_b, two);
+        auto add3 = m1.add_instruction(migraphx::make_op("add"), slice_c, three);
+
+        m1.add_return({add1});    
+    };
+    
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto a = m2.add_parameter("a", as); 
+        auto one = m2.add_literal(
+            migraphx::generate_literal({migraphx::shape::float_type, {1, 77, 768}}, 3.0f));
+        auto two = m2.add_literal(
+            migraphx::generate_literal({migraphx::shape::float_type, {1, 77, 768}}, 2.0f));
+        auto three = m2.add_literal(
+            migraphx::generate_literal({migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+
+        auto concat =
+            m2.add_instruction(migraphx::make_op("concat", {{"axis", 2}}), one, two, three);
+
+        auto add = m2.add_instruction(migraphx::make_op("add"), a, concat);
+
+        auto slice_a = m2.add_instruction(
+            migraphx::make_op("slice", {{"axes", {2}}, {"starts", {0}}, {"ends", {768}}}), add);
+        
+        m2.add_return({slice_a});
+    };    
+    EXPECT(m1.sort() == m2.sort());
+}
+
+TEST_CASE(slice_add_mul)
+{
+    migraphx::shape as{migraphx::shape::float_type, {1, 77, 2304}};
+
+    migraphx::module m1;
+    {
+        auto a = m1.add_parameter("a", as); 
+
+        auto slice_a = m1.add_instruction(
+            migraphx::make_op("slice", {{"axes", {2}}, {"starts", {0}}, {"ends", {768}}}), a);
+        auto slice_b = m1.add_instruction(
+            migraphx::make_op("slice", {{"axes", {2}}, {"starts", {768}}, {"ends", {1536}}}), a);
+        auto slice_c = m1.add_instruction(
+            migraphx::make_op("slice", {{"axes", {2}}, {"starts", {1536}}, {"ends", {2304}}}), a);  
+
+        auto one = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 3.0f));
+        auto two = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 2.0f));
+        auto three = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+        auto four = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+
+        auto add1 = m1.add_instruction(migraphx::make_op("add"), slice_a, one);
+        auto mul1 = m1.add_instruction(migraphx::make_op("mul"), add1, four);
+
+        auto mul2 = m1.add_instruction(migraphx::make_op("mul"), slice_b, two);
+        auto mul3 = m1.add_instruction(migraphx::make_op("mul"), slice_c, three);
+        
+        m1.add_return({mul1, mul2, mul3});    
+    };
+    run_pass(m1);
+    migraphx::module m2 = m1;
+    EXPECT(m1 == m2);
```

Comment:
The `mul` should be fused here.

Reply:
@pfultz2 
I think if we want to fuse `mul` then we will need another matcher like 'slice_add_mul` which will look for slice->add->mul (currently not included in `find_mul_add`).

Example:
Pass1
slice1->add->mul
slice2->mul
slice3->mul

Pass2
slice1->mul->add
slice2->mul
slice3->mul

All the mul will be merged before slicing only if we have `slice_mul_add` matcher.